src/library.cpp

```diff
@@ -27,10 +27,13 @@
 #include "tools/regexTools.h"
 #include "tools/pathTools.h"
 #include "tools/stringTools.h"
+#include "tools/otherTools.h"
```

Comment:
This comment applies to the commit "Use a concurrent_cache to store Archive in library. "

I recommend to split it into three:
- Get rid of `m_readers` in `Library::Impl`
- Switch `Library::Impl::m_archives` to a concurrent cache
- Link the cache size to the book count (unless explicitly configured by the user via the env var).